http-clients/java-nio-client/src/test/java/software/amazon/awssdk/http/nio/java/JavaHttpClientNioAsyncHttpClientWireMockTest.java

```diff
@@ -0,0 +1,302 @@
+
+public class JavaHttpClientNioAsyncHttpClientWireMockTest {
+    private final RecordingNetworkTrafficListener wiremockTrafficListener = new RecordingNetworkTrafficListener();
+
+    @Rule
+    public WireMockRule mockServer = new WireMockRule(wireMockConfig()
+            .dynamicPort()
+            .dynamicHttpsPort()
+            .networkTrafficListener(wiremockTrafficListener));
+
+    private static SdkAsyncHttpClient client = JavaHttpClientNioAsyncHttpClient.builder().build();
+
+    @Before
+    public void methodSetup() {
+        wiremockTrafficListener.reset();
+    }
+
+
+    private void makeSimpleRequest(SdkAsyncHttpClient client) throws Exception {
+        String body = randomAlphabetic(10);
+        URI uri = URI.create("http://localhost:" + mockServer.port());
+        stubFor(any(urlPathEqualTo("/"))
+                .willReturn(aResponse().withBody(body)));
+        SdkHttpRequest request = createRequest(uri);
+        RecordingResponseHandler recorder = new RecordingResponseHandler();
+        client.execute(AsyncExecuteRequest.builder()
+                .request(request)
+                .requestContentPublisher(createProvider(""))
+                .responseHandler(recorder)
+                .build());
+        recorder.completeFuture.get(5, TimeUnit.SECONDS);
+    }
+
+    @Test
+    public void simpleMockTest() throws Exception {
+        try {
+            makeSimpleRequest(client);
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Test
+    public void simpleMockPostTest() throws InterruptedException, ExecutionException, TimeoutException {
+        // Tests whether the HttpClient really only takes content in length of that demanded in "Content-Length"
+        final String content = randomAlphabetic(10);
+        URI uri = URI.create("http://localhost:" + mockServer.port());
+        stubFor(any(urlEqualTo("/echo?reversed=true"))
+                .withRequestBody(equalTo(content))
+                .willReturn(aResponse().withBody(StringUtils.reverse(content))));
+
+        SdkHttpFullRequest request = createRequest(uri, "/echo", content, SdkHttpMethod.POST, singletonMap("reversed", "true"), emptyMap());
+        request = request.toBuilder().putHeader("Content-Length", Integer.toString(content.length())).build();
+        RecordingResponseHandler recorder = new RecordingResponseHandler();
+
+        SdkHttpContentPublisher contentPublisher = createProvider(content);
+        client.execute(AsyncExecuteRequest.builder()
+                .request(request)
+                .requestContentPublisher(contentPublisher)
+                .responseHandler(recorder)
+                .build()).join();
+
+        recorder.completeFuture.get(2, TimeUnit.SECONDS);
+        System.out.println(wiremockTrafficListener.requests + "\n\n\n\n" + wiremockTrafficListener.response);
+
+        assertThat(wiremockTrafficListener.response.toString().endsWith(content));
+
+    }
+
+    // @Test
```

Comment:
This test was abandoned for now since we didn't really check whether the content is no longer than the content-length. I left it here because I thought it could be used later and could remind of the functionality of checking content-length.